src/v2/Apps/Auction2/Routes/Auction2RegistrationRoute.tsx

```diff
@@ -0,0 +1,137 @@
+import { Button, Join, ModalDialog, Spacer } from "@artsy/palette"
+import { createFragmentContainer, graphql } from "react-relay"
+import { Auction2RegistrationRoute_me } from "v2/__generated__/Auction2RegistrationRoute_me.graphql"
+import { Auction2RegistrationRoute_sale } from "v2/__generated__/Auction2RegistrationRoute_sale.graphql"
+import { OnSubmitValidationError } from "v2/Apps/Auction2/Components/Form/OnSubmitValidationError"
+import { Form, Formik } from "formik"
+import { CreditCardInputProvider } from "v2/Components/CreditCardInput"
+import { useRouter } from "v2/System/Router/useRouter"
+import {
+  AuctionFormValues,
+  initialValuesForRegistration,
+  registrationValidationSchema,
+} from "v2/Apps/Auction2/Components/Form/Utils"
+import { useCreateTokenAndSubmit } from "v2/Apps/Auction2/Hooks/useCreateTokenAndSubmit"
+import { useEffect } from "react"
+import { ConditionsOfSaleCheckbox } from "v2/Apps/Auction2/Components/Form/ConditionsOfSaleCheckbox"
+import { AddressFormWithCreditCard } from "v2/Apps/Auction2/Components/Form/AddressFormWithCreditCard"
+import { IdentityVerificationWarning } from "v2/Apps/Auction2/Components/Form/IdentityVerificationWarning"
+import { useAuctionTracking } from "v2/Apps/Auction2/Hooks/useAuctionTracking"
+import { ErrorStatus } from "../Components/Form/ErrorStatus"
+
+interface Auction2RegistrationRouteProps {
+  me: Auction2RegistrationRoute_me
+  sale: Auction2RegistrationRoute_sale
+}
+
+const Auction2RegistrationRoute: React.FC<Auction2RegistrationRouteProps> = ({
+  me,
+  sale,
+}) => {
+  const { tracking } = useAuctionTracking()
+  const { router } = useRouter()
+  const { needsIdentityVerification } = computeProps({ sale, me })
```

Comment:
I don't see how extracting a function here improves things

Reply:
Thinking about your other comment -- in both here and in the file above, the bulk of this is for `needsIdentityVerification`. Def a case for moving to MP. Will do.